app/experimenter/kinto/tasks.py

```diff
@@ -80,16 +80,14 @@ def handle_rejection(kinto_client):
     collection_data = kinto_client.get_rejected_collection_data()
     experiment = NimbusExperiment.objects.get(slug=rejected_slug)
 
-    has_changes = False
     if experiment.status == NimbusExperiment.Status.LIVE and experiment.is_end_requested:
         experiment.is_end_requested = False
-        has_changes = True
-    elif experiment.status == NimbusExperiment.Status.ACCEPTED:
+        experiment.publish_status = NimbusExperiment.PublishStatus.IDLE
+    elif experiment.publish_status == NimbusExperiment.PublishStatus.WAITING:
         experiment.status = NimbusExperiment.Status.DRAFT
-        has_changes = True
 
-    if has_changes:
-        experiment.save()
+    experiment.publish_status = NimbusExperiment.PublishStatus.IDLE
+    experiment.save()
```

Comment:
We removed the `has_changes` check because an experiment arriving here will always be Waiting, and regardless of it hitting either of the conditions above it should be set back to Idle.